pkg/resource/endpoint/hooks.go

```diff
@@ -0,0 +1,209 @@
+package endpoint
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"golang.org/x/exp/slices"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/eventbridge"
+
+	"github.com/aws-controllers-k8s/eventbridge-controller/apis/v1alpha1"
+	"github.com/aws-controllers-k8s/eventbridge-controller/pkg/tags"
+)
+
+const (
+	defaultRequeueDelay = time.Second * 5
+)
+
+var (
+	requeueWaitWhileCreating = ackrequeue.NeededAfter(
+		fmt.Errorf("endpoint in status %q, requeing", svcsdk.EndpointStateCreating),
+		defaultRequeueDelay,
+	)
+
+	requeueWaitWhileUpdating = ackrequeue.NeededAfter(
+		fmt.Errorf("endpoint in status %q, cannot be modified or deleted", svcsdk.EndpointStateUpdating),
+		defaultRequeueDelay,
+	)
+
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		fmt.Errorf("endpoint in status %q, cannot be modified or deleted", svcsdk.EndpointStateDeleting),
+		defaultRequeueDelay,
+	)
+)
+
+type validationError struct {
+	field   string
+	message string
+}
+
+func (v validationError) Error() string {
+	return fmt.Sprintf("invalid Spec: %q: %s", v.field, v.message)
+}
+
+func newValidationError(field, message string) validationError {
+	return validationError{
+		field:   field,
+		message: message,
+	}
+}
```

Comment:
@embano1 a cool technique I use is to create a "base" error using `errors.New()` and then "wrap" the error with a custom message, like this:

```go
import errors

var (
    errValidation = errors.New("Invalid Spec")
)

func newValidationError(field, message string) error {
    return fmt.Errorf("%w: %q: %s", errValidation, field, message)
}
```

Reply:
Oh, that's neat! I'm assuming this works with `errors.Is()` then (whereas my implementation requires `errors.As()`)?